client/extensions/wp-job-manager/state/settings/reducer.js

```diff
@@ -24,6 +27,20 @@ export const fetching = createReducer( {}, {
 } );
 
 /**
+ * Returns the updated saving state after an action has been dispatched.
+ * Saving state tracks whether the settings for a site are currently being saved.
+ *
+ * @param  {Object} state Current saving state
+ * @param  {Object} action Action object
+ * @return {Object} Updated saving state
+ */
+export const saving = createReducer( {}, {
+	[ WP_JOB_MANAGER_SAVE_SETTINGS ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: true } ),
+	[ WP_JOB_MANAGER_SAVE_SUCCESS ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+	[ WP_JOB_MANAGER_SAVE_ERROR ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+} );
```

Comment:
Please note that we have a helpful utility method to build these kinds of reducers in a less convoluted way…

```js
import { keyedReducer } from 'state/utils';

export const isSavingSite = ( state = false, { type } ) => get( {
	[ WP_JOB_MANAGER_SAVE_SETTINGS ]: true,
	[ WP_JOB_MANAGER_SAVE_SUCCESS ]: false,
	[ WP_JOB_MANAGER_SAVE_SUCCESS ]: false,
}, type, state );

export const isSaving = keyedReducer( 'siteId', isSavingSite )
```

 - the naming of `isSaving` matches its type (boolean)
 - the noise of updating based on a key has been removed
 - we can test in isolation without rigging up options
 - the default state type matches the actual state type (state here is a boolean, not an object)